spring-cloud-gcp-autoconfigure/src/main/java/com/google/cloud/spring/autoconfigure/pubsub/GcpPubSubAutoConfiguration.java

```diff
@@ -535,15 +538,14 @@ private ExecutorProvider createAndRegisterExecutorProvider(String beanName, Thre
 		return executor;
 	}
 
-	private boolean hasSameGlobalAndSelectiveSettings(String subscriptionName) {
-		Integer globalNumThreads = getGlobalExecutorThreads();
-		Integer selectiveNumThreads = this.gcpPubSubProperties
-				.getSubscriber(subscriptionName, this.finalProjectIdProvider.getProjectId()).getExecutorThreads();
-		return selectiveNumThreads.equals(globalNumThreads);
-	}
-
 	private Integer getGlobalExecutorThreads() {
 		Integer numThreads = this.gcpPubSubProperties.getSubscriber().getExecutorThreads();
 		return numThreads != null ? numThreads : PubSubConfiguration.DEFAULT_EXECUTOR_THREADS;
 	}
+
+	private List<String> getNameAndFullPath(String subscriptionName) {
+		ProjectSubscriptionName subscriptionReference = PubSubSubscriptionUtils
+				.toProjectSubscriptionName(subscriptionName, this.finalProjectIdProvider.getProjectId());
+		return Arrays.asList(subscriptionReference.getSubscription(), subscriptionReference.toString());
```

Comment:
My first thought was that this is a job for a pair, but I realized that `ProjectSubscriptionName` already encapsulates both pieces of information. So we can just use the `ProjectSubscriptionName` object everywhere where `getNameAndFullPath()` is used. 
This helper might be nice to keep, although perhaps renamed to `getFullSubscriptionName` or similar.